### PR TITLE
Create specific recipe page

### LINKED ISSUE #5: Add specific recipe page
Ideally done after #4, but can be done without the abstraction I guess
#9 will add the endpoint on Recipes API for retrieving a specific recipe, so until then, you can just get all recipes and let id be the index of that list 

On the front end, when navigated to `/recipes/:id`, a user should be presented a page with details about a specific recipe. For now, just output all of the data of the recipe (we'll come up with a better design/presentation down the line).

Please handle case in UI when the recipe can't be found by id.

in `/recipes` (the page that lists all recipe titles) make the titles links that would take you to the recipe's page.

Some [routing docs](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-8.0) to start off

### PR DESCRIPTION
**Pull request by 6ar8nas**
### Notes

Prerequisite #15, waiting for it to be merged before finalizing my changes.

### Demo

Full size:
Screenshot 2024-05-11 at 23 50 42

Mobile:
Screenshot 2024-05-11 at 23 51 15

Closes #5

### CODE REVIEW

**Comment by adomas-vensas** [comment]
If we are to have `Ingredients` not as standalone entities but as simple strings, I think the alignment of the `Ingredients` should stay to the left because, otherwise, you would have to parse the ingredient data. I don't think that would be easy, considering that quantities and the method of preparing the ingredients can vary significantly.

So from this:

image

To this:

image

**Comment by 6ar8nas** [comment]
Yeah, as I've noted I won't be merging this before the entities are finalized, so I'll address the changes needed after the prerequisite #15 is complete.

File: FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+<style>
+    @@media 
+</style>
+
+@if (_task is not { IsCompleted: true })
+{
+    <PageTitle>Recipe loading</PageTitle>
+    <FluentProgressRing />
+}
```
Comment (BenasB) [inline FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor:16]: If you're feeling fancy, we could throw in some skeletons while it's loading: https://fluentui-blazor.net/Skeleton

Comment (6ar8nas) [reply FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor]: I've put some simplified version of the page as a skeleton.

File: FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            <FluentStack Orientation="Orientation.Vertical" Style="min-width: 500px;">
+                <FluentOverflow>
+                    @foreach (var item in _tagNames)
+                    {
+                        <FluentOverflowItem><FluentBadge Appearance="Appearance.Accent">@item</FluentBadge></FluentOverflowItem>
+                    }
+                </FluentOverflow>
+                <p>@_description</p>
+                <FluentLabel Color="Color.Accent" Typo="Typography.H4">Ingredients</FluentLabel>
+                <FluentDataGrid Items=_ingredients.AsQueryable() GridTemplateColumns="2fr 1fr" Style="width: 100%;" ShowHover GenerateHeader="GenerateHeaderOption.None">
```
Comment (BenasB) [inline FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor:55]: We have render instructions without `FluentDataGrid` maybe this should also do the same?

File: FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    <FluentLabel Typo="Typography.Body" MarginBlock="10px">The requested recipe doesn't exist.</FluentLabel>
+}
+else
+{
+    <div style="height: 14rem; overflow: hidden; display: flex; justify-content: center; align-items: center;">
+        <img src="https://worldfoodtour.co.uk/wp-content/uploads/2013/06/neptune-placeholder-48.jpg" alt="@_recipe.Title"/>
+    </div>
+    <FluentStack Style="margin-block: 10px;">
+        <FluentLabel Typo="Typography.H1" Style="flex-grow: 1">@_recipe.Title</FluentLabel>
+        <FluentStack Width="auto">
```
Comment (BenasB) [inline FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor:39]: [`<FluentSpacer/>`](https://www.fluentui-blazor.net/Spacer)? (From source code only, not sure if this makes them align to the right, or `Style="flex-grow: 1"` but the point still stands)

Comment (6ar8nas) [reply FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor]: Yeah, that works. Addressed.

File: FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor (hunk 2 above)
Comment (6ar8nas) [reply FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor]: We could. I feel like this being different makes the page look somewhat better though.

**Comment by 6ar8nas** [comment]
Updated the demo screenshots.

File: FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                    <li style="margin-block: 10px;">@instruction</li>
+                }
+            </ol>
+        </FluentGridItem>
+    </FluentGrid>
+}
+
+@code {
+    [Parameter]
+    public int Id { get; set; }
```
Comment (BenasB) [inline FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor:85]: NIT: probably init instead of set

Comment (6ar8nas) [reply FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor:85]: You're right, though this will be overrided by the prerequisite.

**Pull request merged**